crates/containerd-shim-wasm/src/sandbox/containerd.rs

```diff
@@ -62,41 +129,268 @@ impl Client {
         })
     }
 
-    pub fn get_image_content_sha(&self, image_name: impl ToString) -> Result<String> {
+    // used in tests to clean up content
+    #[allow(dead_code)]
+    fn delete_content(&self, digest: impl ToString) -> Result<()> {
+        self.rt.block_on(async {
+            let req = DeleteContentRequest {
+                digest: digest.to_string(),
+            };
+            let req = with_namespace!(req, self.namespace);
+            ContentClient::new(self.inner.clone())
+                .delete(req)
+                .await
+                .map_err(|err| ShimError::Containerd(err.to_string()))?;
+            Ok(())
+        })
+    }
+
+    // wrapper around lease that will create a lease and return a guard that will delete the lease when dropped
+    fn lease(&self, r#ref: String) -> Result<LeaseGuard> {
+        self.rt.block_on(async {
+            let mut lease_labels = HashMap::new();
+            let expire = chrono::Utc::now() + chrono::Duration::hours(24);
+            lease_labels.insert("containerd.io/gc.expire".to_string(), expire.to_rfc3339());
+            let lease_request = containerd_client::services::v1::CreateRequest {
+                id: r#ref.clone(),
```

Comment:
The reference gets moved below on line 167, so this clone is needed 

```
borrow of moved value: `reference`
move occurs because `reference` has type `std::string::String`, which does not implement the `Copy` 
```

Reply:
i removed the extra clone in the error message below